docs_src/microservices/application/Triggers.md

```diff
@@ -1,109 +1,198 @@
-        ConnectTimeout = "5" # Seconds
-        SkipCertVerify = "false"
-        CertFile       = ""
-        KeyFile        = ""
-        KeyPEMBlock    = ""
-        CertPEMBlock   = ""
+    [Trigger.EdgexMessageBus.SubscribeHost]
+    Host = "localhost"
+    Port = 1883
+    Protocol = "tcp"
+    SubscribeTopics="edgex/events/#"
+    [Trigger.EdgexMessageBus.PublishHost]
+    Host = "localhost"
+    Port = 1883
+    Protocol = "tcp"        
+    PublishTopic="" # optional if publishing response back to the MessageBus
+    [Trigger.EdgexMessageBus.Optional]
+    # MQTT Specific options
+    ClientId ="new-app-service"
+    Qos            = "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive      = "10" # Seconds (must be 2 or greater)
+    Retained       = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "30" # Seconds
+    SkipCertVerify = "false"
+    authmode = 'none'  # change to 'usernamepassword', 'clientcert', or 'cacert' for secure MQTT messagebus.
+    secretname = 'mqtt-bus'
+
+```
+
+
+
+!!! edgey "EdgeX 2.0"
+    New for EdgeX 2.0 is the Secure MessageBus when use the `Redis Pub/Sub` implementation. See the [Secure MessageBus](../../security/Ch-Secure-MessageBus.md) documentation for more details.
+
+!!! edgey "EdgeX 2.0"
+    Also new for EdgeX 2.0 is the MQTT MessageBus implementation now supports retrieving secrets from the `Secret Store` for secure MQTT connection, but there is not any facility yet to generate the credentials on first startup and distribute them to all services, as is done with `Redis Pub/sub`. This MQTT credentials generation and distribution is a future enhancement for EdgeX security services. 
+
+### Filter By Topics
+
+!!! edgey "EdgeX 2.0"
+    New for EdgeX 2.0
+
+App services now have the capability to filter by EdgeX MessageBus topics rather then using Filter functions in the functions pipeline. Filtering by topic is more efficient since the App Service never receives the data off the MessageBus. Core Data and/or Device Services now publish to multi-level topics that include the `profilename`, `devicename` and `sourcename` . Sources are the `commandname` or `resourcename` that generated the Event. The publish topics now look like this:
+
+```
+# From Core Data
+edgex/events/core/<profile-name>/<device-name>/<source-name>
+
+# From Device Services
+edgex/events/device/<profile-name>/<device-name>/<source-name>
 ```
 
+This with App Services capability to have multiple subscriptions allows for multiple filters by subscriptions. The `SubscribeTopics` setting takes a comma separated list of subscribe topics.
+
+ Here are a few examples of how to configure the `SubscribeTopics` setting under the `Trigger.EdgexMessageBus.SubscribeHost` section to filter by subscriptions using the `profile`, `device` and `source` names from the Virtual Device Service:
+
+- Filter for all Events 
+
+  ```toml
+  SubscribeTopics="edgex/events/#
```

Comment:
In general I think it makes sense here to use a device name and profile name that are easier to differentiate, even if it is "my-device-profile" and "my-device-name" in this section.

Reply:
Will use the Device SNMP names which have better names for this example.